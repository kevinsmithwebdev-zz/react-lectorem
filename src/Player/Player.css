audio:focus {
  outline: none;
}

audio {
  filter: invert(20%);
  width: 400px;
  height: 50px;
}
