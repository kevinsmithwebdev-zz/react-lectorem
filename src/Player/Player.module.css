audio {
  filter: invert(20%);
  width: 400px;
  height: 50px;
}

audio:focus {
  outline: none;
}
