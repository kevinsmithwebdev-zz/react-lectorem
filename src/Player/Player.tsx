import React from 'react';
import ReactAudioPlayer from 'react-audio-player';
import classes from './Player.module.css';

const LISTEN_INTERVAL_IN_MS = 10;

interface PlayerInterface {
  audioPath: string,
  onListen: any, // FIXME:
}

const Player: React.FC<PlayerInterface> = ({ onListen, audioPath }): JSX.Element => (
  <div className={ classes.Player }>
    <ReactAudioPlayer
      listenInterval={ LISTEN_INTERVAL_IN_MS }
      onListen={ onListen }
      src={ audioPath }
      controlsList='nodownload'
      controls
    />
  </div>
);

export default Player;
