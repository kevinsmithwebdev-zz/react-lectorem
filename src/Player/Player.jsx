import React from "react";
import ReactAudioPlayer from 'react-audio-player';
import './Player.css';

const LISTEN_INTERVAL_IN_MS = 10;

const Player = props => {
  return (
    <div className='Player'>
      <h2>Player</h2>
      <ReactAudioPlayer
        listenInterval={ LISTEN_INTERVAL_IN_MS }
        onListen={ props.onListen }
        src={ props.audioPath }
        controlsList='nodownload'
        controls
      />
    </div>
  );
}

export default Player;
