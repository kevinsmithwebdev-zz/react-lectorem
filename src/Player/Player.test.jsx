import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Player from './Player';

describe('Player', () => {
  describe('the main component render', () => {
    const renderer = new ShallowRenderer();
    describe('empty', () => {
      it('should render without crashing', () => {
        renderer.render(<Player />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
      });
    });

    describe('with data', () => {
      const onListen = () => {};
      const props = { onListen, audioPath: 'audioPath123' };
      it('should render and match snapshot', () => {
        renderer.render(<Player { ...props } />);
        expect(renderer.getRenderOutput()).toMatchSnapshot();
      });
    });
  });
});
