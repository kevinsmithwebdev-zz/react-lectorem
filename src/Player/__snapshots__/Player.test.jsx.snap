// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player the main component render empty should render without crashing 1`] = `
<div
  className="Player"
>
  <ReactAudioPlayer
    autoPlay={false}
    className=""
    controls={true}
    controlsList="nodownload"
    id=""
    listenInterval={10}
    loop={false}
    muted={false}
    onAbort={[Function]}
    onCanPlay={[Function]}
    onCanPlayThrough={[Function]}
    onEnded={[Function]}
    onError={[Function]}
    onListen={[Function]}
    onLoadedMetadata={[Function]}
    onPause={[Function]}
    onPlay={[Function]}
    onSeeked={[Function]}
    onVolumeChanged={[Function]}
    preload="metadata"
    style={Object {}}
    title=""
    volume={1}
  />
</div>
`;

exports[`Player the main component render with data should render and match snapshot 1`] = `
<div
  className="Player"
>
  <ReactAudioPlayer
    autoPlay={false}
    className=""
    controls={true}
    controlsList="nodownload"
    id=""
    listenInterval={10}
    loop={false}
    muted={false}
    onAbort={[Function]}
    onCanPlay={[Function]}
    onCanPlayThrough={[Function]}
    onEnded={[Function]}
    onError={[Function]}
    onListen={[Function]}
    onLoadedMetadata={[Function]}
    onPause={[Function]}
    onPlay={[Function]}
    onSeeked={[Function]}
    onVolumeChanged={[Function]}
    preload="metadata"
    src="audioPath123"
    style={Object {}}
    title=""
    volume={1}
  />
</div>
`;
